manifests/init.pp

```diff
@@ -185,26 +185,26 @@
   Optional[Stdlib::Absolutepath] $ssl_password_file          = undef,
 
   ### START Package Configuration ###
-  $package_ensure                                            = present,
-  $package_name                                              = $nginx::params::package_name,
-  $package_source                                            = 'nginx',
-  $package_flavor                                            = undef,
+  String[1] $package_ensure                                  = present,
+  String[1] $package_name                                    = $nginx::params::package_name,
+  String[1] $package_source                                  = 'nginx',
+  Optional[String[1]] $package_flavor                        = undef,
   Boolean $manage_repo                                       = $nginx::params::manage_repo,
   Hash[String[1], String[1]] $mime_types                     = $nginx::params::mime_types,
   Boolean $mime_types_preserve_defaults                      = false,
-  Optional[String] $repo_release                             = undef,
-  $passenger_package_ensure                                  = 'present',
+  Optional[String[1]] $repo_release                          = undef,
+  String[1] $passenger_package_ensure                        = 'present',
   String[1] $passenger_package_name                          = $nginx::params::passenger_package_name,
   Optional[Stdlib::HTTPUrl] $repo_source                     = undef,
   ### END Package Configuration ###
 
   ### START Service Configuation ###
-  $service_ensure                                            = running,
-  $service_enable                                            = true,
-  $service_flags                                             = undef,
-  $service_restart                                           = undef,
-  $service_name                                              = 'nginx',
-  $service_manage                                            = true,
+  Enum['running', 'absent', 'stopped', 'undef'] $service_ensure = running,
```

Comment:
what's undef doing here?

Reply:
Copied from the previous PR. Haven't looked too closely at the content yet.